core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -618,7 +618,7 @@ void core_mmu_create_user_map(struct user_ta_ctx *utc,
 	memset(dir_info.table, 0, PGT_SIZE);
 	core_mmu_populate_user_map(&dir_info, utc);
 	map->user_map = virt_to_phys(dir_info.table) | TABLE_DESC;
-	map->asid = utc->context & TTBR_ASID_MASK;
+	map->asid = utc->mmu->asid;
```

Comment:
This should check ASID do not exeed hw limit (32).
As this is void routine, maybe ` COMPILE_TIME_ASSERT(MMU_NUM_ASIDS <= 2 * 32);` with an exported `MMU_NUM_ASIDS`.

Or tee_mmu.c to check `MMU_NUM_ASIDS` against lpae/non-lpae heander file.
Or tee_mmu.c to comment `MMU_NUM_ASIDS` with `/* 64 = 2 * 32, matches all mappings supported by core */`.

Reply:
Hardware limit of 32? Do you have a reference?

Reply:
Current AArch32 and AArch64 support up to 32 value for the ASID, no?

Reply:
As far as I know it can be up to 8 bits in AArch32.

Reply:
right. 8bit, 256 values for both 32bit mmu, and AArch32/AArch64 LPAE.

Reply:
Resolved in **tee_mmu.c**. thanks.